game/scripts/vscripts/util.ts

```diff
@@ -185,6 +185,68 @@ export function removeHighlight(path: string) {
     CustomGameEventManager.Send_ServerToAllClients("remove_highlight", { path });
 }
 
+/**
+ * Returns the string to highlight in the UI
+ * @param item The item you want to highlight
+ */
+function getItemInventoryPathString(item:CDOTA_Item): string|undefined {
+    const hero = getOrError(getPlayerHero());
+    let slot = undefined;
+
+    if (hero.GetItemInSlot(InventorySlot.NEUTRAL_SLOT) === item ) {
+        return "HUDElements/lower_hud/center_with_stats/inventory_composition_layer_container/inventory_neutral_slot_container/inventory_neutral_slot";
+    }
+
+    for (let i = 0; i < DOTA_ITEM_INVENTORY_SIZE; i++) {
+        if (hero.GetItemInSlot(i) === item) {
+            slot = i;
+            break;
+        }
+    }
+    
+    if (slot === undefined) {
+        return;
+    }
+    if (slot <= InventorySlot.SLOT_3) {
+        return "HUDElements/lower_hud/center_with_stats/center_block/inventory/inventory_items/InventoryContainer/inventory_list_container/inventory_list/inventory_slot_" + slot;
+    } else if (slot <= InventorySlot.SLOT_6) {
+        return "HUDElements/lower_hud/center_with_stats/center_block/inventory/inventory_items/InventoryContainer/inventory_list_container/inventory_list2/inventory_slot_" + slot;
+    } else if (slot <= InventorySlot.SLOT_9) {
+        return "HUDElements/lower_hud/center_with_stats/center_block/inventory/inventory_items/InventoryContainer/inventory_list_container/inventory_backpack_list/inventory_slot_" + slot;
+    }
+}
+
+/**
+ * Highlights the slot of an item if it exists, returns the string for you to remove later
+ * @param item the item you want to highlight
+ */
+export function highlightItemInUi(item:CDOTA_Item) {
+    const pathString = getItemInventoryPathString(item);
+    if (pathString !== undefined) {
+        highlightUiElement(pathString!);
+    }
+    
+    return pathString
+}
+
+/**
+ * Checks if a point is inside an array of points
+ * @param point The point to check
+ * @param polygon The array of points to check against
+ */
+export function isPointInsidePolygon(point: Vector, polygon: Vector[]) {
```

Comment:
Thought this was no longer used in the new CH3 and could be removed?

Reply:
No references, so I yea. it certainly shouldnt be added here.